src/System.Security.Cryptography.X509Certificates/tests/InteropTests.cs

```diff
@@ -0,0 +1,329 @@
+                    bool success = CryptQueryObject(
+                        CertQueryObjectType.CERT_QUERY_OBJECT_BLOB,
+                        ref certBlob,
+                        ExpectedContentTypeFlags.CERT_QUERY_CONTENT_FLAG_CERT,
+                        ExpectedFormatTypeFlags.CERT_QUERY_FORMAT_FLAG_BINARY,
+                        0,
+                        IntPtr.Zero,
+                        IntPtr.Zero,
+                        IntPtr.Zero,
+                        IntPtr.Zero,
+                        IntPtr.Zero,
+                        out pCertContext
+                            );
+
+                    if (!success)
+                    {
+                        int hr = Marshal.GetHRForLastWin32Error();
+                        throw new CryptographicException(hr);
+                    }
+                }
+            }
+
+            // Now, create an X509Certificate around our handle.
+            using (X509Certificate2 c = new X509Certificate2(pCertContext))
+            {
+                // And release our ref-count on the handle. X509Certificate better be maintaining its own.
+                CertFreeCertificateContext(pCertContext);
+
+                // Now, test various properties to make sure the X509Certificate actually wraps our CERT_CONTEXT.
+                IntPtr h = c.Handle;
+                Assert.Equal(pCertContext, h);
+                pCertContext = IntPtr.Zero;
+
+                string issuer = c.Issuer;
+                Assert.Equal(
+                    "CN=Microsoft Code Signing PCA, O=Microsoft Corporation, L=Redmond, S=Washington, C=US",
+                    issuer);
+
+                byte[] expectedPublicKey = (
+                    "3082010a0282010100e8af5ca2200df8287cbc057b7fadeeeb76ac28533f3adb" +
+                    "407db38e33e6573fa551153454a5cfb48ba93fa837e12d50ed35164eef4d7adb" +
+                    "137688b02cf0595ca9ebe1d72975e41b85279bf3f82d9e41362b0b40fbbe3bba" +
+                    "b95c759316524bca33c537b0f3eb7ea8f541155c08651d2137f02cba220b10b1" +
+                    "109d772285847c4fb91b90b0f5a3fe8bf40c9a4ea0f5c90a21e2aae3013647fd" +
+                    "2f826a8103f5a935dc94579dfb4bd40e82db388f12fee3d67a748864e162c425" +
+                    "2e2aae9d181f0e1eb6c2af24b40e50bcde1c935c49a679b5b6dbcef9707b2801" +
+                    "84b82a29cfbfa90505e1e00f714dfdad5c238329ebc7c54ac8e82784d37ec643" +
+                    "0b950005b14f6571c50203010001").HexToByteArray();
+
+                byte[] publicKey = c.GetPublicKey();
+                Assert.Equal(expectedPublicKey, publicKey);
+
+                byte[] expectedThumbPrint = "108e2ba23632620c427c570b6d9db51ac31387fe".HexToByteArray();
+                byte[] thumbPrint = c.GetCertHash();
+                Assert.Equal(expectedThumbPrint, thumbPrint);
+            }
+        }
+
+        [DllImport("crypt32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
```

Comment:
This interop code is duplicative of what's used by the product, right?  We're doing that to avoid the tests taking a dependency on it in case it's the source of the problem?